rust/src/cli/service.rs

```diff
@@ -99,3 +111,86 @@ fn relocate_file(file_path: &Path) -> Result<(), CliError> {
     );
     Ok(())
 }
+
+fn pin_iface_name(cfg_dir: &Path) -> Result<(), CliError> {
+    let file_path = cfg_dir.join(PIN_STATE_FILENAME);
+    let mut fd = std::fs::File::open(&file_path)?;
+    let mut content = String::new();
+    fd.read_to_string(&mut content)?;
```

Comment:
You almost always want to use a `BufReader` when reading from files.  This is actually just shorter and faster as e.g.:

```
    let content = std::fs::read_to_string(&file_path)?;
```